pkg/utils/kubernetes/ownerrefs.go

```diff
@@ -38,6 +42,40 @@ func SetOwnerForObject(obj, owner client.Object) {
 	}
 }
 
+// SetOwnerForObjectThroughLabels sets the owner of the provided object through a label.
+func SetOwnerForObjectThroughLabels(obj, owner client.Object) error {
+	labels := obj.GetLabels()
+	managedByLabelSet, err := GetManagedByLabelSet(owner)
+	if err != nil {
+		return err
+	}
+	for k, v := range managedByLabelSet {
+		labels[k] = v
+	}
+	obj.SetLabels(labels)
+	return nil
+}
+
+// GetManagedByLabelSet returns a map of labels with the managing object's metadata.
```

Comment:
```suggestion
// GetManagedByLabelSet returns a map of labels with the provided object's metadata.
// These can be applied to other objects that are owned by the object provided as an argument.
```

WDYT about this wording?